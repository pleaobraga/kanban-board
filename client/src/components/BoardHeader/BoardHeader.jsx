import './BoardHeader.scss'

import React from 'react'
import PropTypes from 'prop-types'
import { startCase } from 'lodash'

export const BoardHeader = ({ name }) => {
  const [createButton, setCreateButton] = React.useState(false)
  const [newCard, setNewCard] = React.useState({
    type: '',
    severity: '',
    duration: ''
  })

  const cardTask = {
    types: ['feature', 'bugFix', 'update', 'research', 'content'],
    severities: ['hight', 'medium', 'low']
  }

  const handleChange = event => {
    event.preventDefault()

    setNewCard({
      ...newCard,
      [event.target.name]: event.target.value
    })
  }

  const createCardTask = () => {}

  return (
    <div className="board-header">
      <h1 className="board-header--title"> Board Name: {startCase(name)}</h1>
      {createButton ? (
        <form className="form" id="createButton">
          <h2>New Task:</h2>

          <select name="type" value={newCard.type} onChange={handleChange}>
            {cardTask.types.map(type => (
              <option key={type} value={type}>
                {type}
              </option>
            ))}
          </select>

          <select
            name="severity"
            value={newCard.severity}
            onChange={handleChange}
          >
            {cardTask.severities.map(severity => (
              <option key={severity} value={severity}>
                {severity}
              </option>
            ))}
          </select>

          <input
            name="severity"
            value={newCard.severity}
            onChange={handleChange}
          />

          <button onClick={createCardTask}>Create Task</button>
          <button>Cancel</button>
        </form>
      ) : (
        <div className="board-header--actions">
          <button onClick={() => setCreateButton(true)}>Add new task </button>
        </div>
      )}
    </div>
  )
}

BoardHeader.propTypes = {
  name: PropTypes.string.isRequired
}

export default BoardHeader
