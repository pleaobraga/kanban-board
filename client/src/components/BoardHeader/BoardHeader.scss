@import '../../styles/_color.scss';

.board-header {
  display: flex;
  flex-direction: column;
  align-items: center;

  margin: 5rem;

  &__title {
    font-size: 8rem;
  }

  .form {
    display: flex;
    flex-direction: column;

    &__title {
      font-size: 2.5rem;
    }

    &__values {
      display: flex;
      flex-wrap: wrap;

      margin: 1.3rem 0;

      .form-control {
        margin-top: 3px;

        label {
          margin-right: 11px;
          font-size: 2rem;
        }

        select,
        input {
          background: $white;
          font-size: 2rem;
          margin-right: 11px;
        }

        input {
          width: 7rem;
        }
      }
    }

    &__actions {
      .button {
        margin-right: 11px;
      }
    }
  }

  .button {
    padding: 1rem;
    font-size: 2rem;
    font-weight: 500;

    border: none;
    border-radius: 4px;

    line-height: 1.75;
    letter-spacing: 0.02857em;
    text-transform: uppercase;

    background-color: #e0e0e0;

    box-sizing: border-box;
    box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2),
      0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 1px 5px 0px rgba(0, 0, 0, 0.12);

    &--primary {
      color: $white;
      background-color: $blue;
      font-size: 1.5rem;
    }

    &--secondary {
      color: $white;
      background-color: $red;
      font-size: 1.5rem;
    }

    &--add {
      background: $green;
      color: $white;
    }
  }
}
