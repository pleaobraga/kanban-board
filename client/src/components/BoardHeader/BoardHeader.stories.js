import React from 'react'
import { storiesOf } from '@storybook/react'

import ProviderWrapper from '../../../.storybook/provider'
import store from '../../../.storybook/configureStore'
import { board } from '../../reducers/__mocks__/reduxMock'
import { BoardHeader } from './BoardHeader'

const initialState = {
  board,
  loadingContent: false,
  errorContent: false
}

const newStore = store(initialState)

const withProvider = story => (
  <ProviderWrapper store={newStore}>{story()}</ProviderWrapper>
)

export default storiesOf('Components | Board Header', module)
  .addDecorator(withProvider)
  .add('default', () => <BoardHeader name={'KanbanBoard'} />, {
    info: {
      inline: true,
      header: false,
      source: false,
      propTables: [BoardHeader]
    }
  })
