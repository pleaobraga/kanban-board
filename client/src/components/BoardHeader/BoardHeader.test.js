import React from 'react'
import { mount } from 'enzyme'
import BoardHeader from './BoardHeader'
import { Provider } from 'react-redux'
import {
  mockStore,
  initialStateRootReducer
} from '../../reducers/__mocks__/reduxMock'

const setup = ({ state = {} }) => {
  const newState = {
    board: {
      ...initialStateRootReducer.content,
      ...state
    }
  }

  const store = mockStore(newState)

  const contentPage = mount(
    <Provider store={store}>
      <BoardHeader name="test" />
    </Provider>
  )

  return contentPage
}

describe('Board Header', () => {
  it('should render properly', () => {
    const component = setup({})
    expect(component).toMatchSnapshot()
  })
})
