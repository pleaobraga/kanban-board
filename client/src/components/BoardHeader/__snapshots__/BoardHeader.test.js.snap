// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Header should render properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BoardHeader
    name="test"
  >
    <div
      className="board-header"
    >
      <h1
        className="board-header__title"
      >
         Board Name: 
        Test
      </h1>
      <div
        className="board-header__actions"
      >
        <button
          className="button button--add"
          onClick={[Function]}
        >
          Add new task
           
        </button>
      </div>
    </div>
  </BoardHeader>
</Provider>
`;
